### DISCUSSION TITLE
Text Generation Library

### DISCUSSION THREAD

**Discussion by samgoldman** [hannobraun/fornjot#2201] [General]
Hi! I've been tinkering with fornjot recently and ended up creating a library that enables creating text with fornjot. It's still very rough, but would you be okay with me uploading it to a separate repository and publishing it as `fj-text` or something similar?

![image](https://github.com/hannobraun/fornjot/assets/6364482/1f371c35-51aa-4714-b9ee-c2f6e2b590e8)

**Comment by hannobraun** [comment]
Hey @samgoldman, that looks awesome! Absolutely, please feel free to publish! `fj-text` seems like a good name to me.

Please post a link, once it's public. I'd love to should it out on Fornjot's social media channels!

**Reply by samgoldman** [reply]
Crate here: https://crates.io/crates/fj-text
Repository here: https://github.com/samgoldman/fj-text

**Reply by hannobraun** [reply]
Thanks! I shouted it out on Matrix, Mastodon, and Twitter!

Suggestion: Maybe consider publishing it under a version number that matches that of Fornjot? Fornjot is still far from being stable, and future releases are sure to break `fj-text`. If your version number matches that of the Fornjot crates, it's easy for users to see what is compatible.

On the other hand, that would tie your release schedule to that of Fornjot, which might or might not matter. Not sure if that's the right trade-off.

**Reply by samgoldman** [reply]
I missed this, sorry! Thanks for the shout out (and I am the user you tagged on Matrix, I'm just new to that platform and didn't see it there either)! I think I'll consider tying the version number to Fornjot once I'm more confident in the API for `fj-text`. Right now I'm still figuring out what works best, but haven't had a ton of time to think about it yet.

**Reply by hannobraun** [reply]
Yeah, makes sense!
